src/core/server/http/integration_tests/http_service.test.ts

```diff
@@ -76,7 +76,7 @@ describe('http service', () => {
           .get(root, legacyUrl)
           .expect(200, 'ok from legacy server');
 
-        expect(response.header['set-cookie']).toBe(undefined);
+        expect(response.header['set-cookie']).toHaveLength(1);
```

Comment:
Why did this change? Is it related to the security plugin registering cookie behaviors? If so, I think we need another way to handle this because people should be able to run tests against purely OSS stuff.

Reply:
I believe it's still "OSS-compliant" and depends on `await http.registerAuth...` call above. This `auth` handler wasn't processing legacy URLs before, but now it does since we use the same Hapi instance. But we definitely need a clarification comment for this :)

Reply:
> I believe it's still "OSS-compliant" and depends on await http.registerAuth... call above

right

> But we definitely need a clarification comment for this

Before Hapi did store a fresh cookie in an internal state of the request object and when we did `h.abandon` в Legacy proxy that state was lost. And Now the cookie is set because of we the common hapi instance.